airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -671,6 +680,15 @@ def run_query(self,
                     self.log.info("Adding experimental 'schemaUpdateOptions': "
                                   "{0}".format(schema_update_options))
 
+                if param_name == 'clustering' and param:
+                    # given the ordering of the query_params_list above,
+                    # if time_partitioning is configured, it will now be in the
+                    # query resource
+                    if 'timePartitioning' not in configuration['query']:
```

Comment:
I'd prefer removing client-side validation such as this, since the BigQuery API may change this and similar restrictions in the future.